Sources/SwiftParser/Declarations.swift

```diff
@@ -2023,9 +2038,29 @@ extension Parser {
   /// =======
   ///
   /// macro-expansion-declaration → '#' identifier expr-call-suffix?
-  mutating func parseMacroExpansionDeclaration() -> RawMacroExpansionDeclSyntax {
-    let poundKeyword = self.consumeAnyToken()
-    let (unexpectedBeforeMacro, macro) = self.expectIdentifier()
+  mutating func parseMacroExpansionDeclaration(
+    _ attrs: DeclAttributes,
+    _ handle: RecoveryConsumptionHandle
+  ) -> RawMacroExpansionDeclSyntax {
+
+    let (unexpectedBeforePound, poundKeyword) = self.eat(handle)
+    // Don't allow space between '#' and the macro name.
+    if poundKeyword.trailingTriviaByteLength != 0 || self.currentToken.leadingTriviaByteLength != 0 {
+      return RawMacroExpansionDeclSyntax(
```

Comment:
The intent here is that parse `#` followed by whitespaces as `#<missing identifier>`, and let the caller do the recovery.

Reply:
I think we should still continue macro attribute parsing if the identifier is at the same line as the `#`. I would expect that to yield better recovery.

Reply:
What would you expect as the parsed tree. How do we embed "hasError"?

Reply:
For example if you have `# elif`, we should have `unexpectedBeforeMacroName = ' elif'` and `macroName` should be a missing identifier with text `elif`. That’s also what we do when parsing `a. b` (where whitespace around the `.` is not balanced).

Reply:
Thanks. Will do.